### PR TITLE
b/249381233: add a flag "backend_cluster_maximum_requests:

### PR DESCRIPTION
**Pull request by qiwzhang**
Signed-off-by: Wayne Zhang <qiwzhang@google.com>

To config envoy cluster circuit breaker `cluster max requests` for all backend clusters.

### CODE REVIEW

File: tests/integration_test/backend_circuit_breaker_test/backend_circuit_breaker_test.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+					api_key: "this-is-an-api-key"
+					auth_token: "this-is-auth-token"
+				}
+			}
+		}
+	}
+}`
+
+	// expected error message from envoy
+	overflowError = "upstream connect error or disconnect/reset before headers. reset reason: overflow"
```
Comment (shuoyang2016) [inline tests/integration_test/backend_circuit_breaker_test/backend_circuit_breaker_test.go:51]: nit: add this into the tests params.

File: tests/integration_test/backend_circuit_breaker_test/backend_circuit_breaker_test.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+			result, err := client.RunGRPCEchoTest(testPlan, s.Ports().ListenerPort)
+			if err == nil {
+				if tc.wantError {
+					t.Errorf("expected error, but not error.")
+				}
+				return
+			}
+
+			if !tc.wantError {
+				t.Errorf("fail, error during running test: %v", err)
```
Comment (shuoyang2016) [inline tests/integration_test/backend_circuit_breaker_test/backend_circuit_breaker_test.go:92]: fail, error during running test: %v; want no error

File: tests/integration_test/backend_circuit_breaker_test/backend_circuit_breaker_test.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+			} else if !strings.Contains(result, overflowError) {
+				t.Errorf("diff error message, got: %v, expect: %s", result, overflowError)
```
Comment (shuoyang2016) [inline tests/integration_test/backend_circuit_breaker_test/backend_circuit_breaker_test.go:94]: s/%s/%q

**Review by shuoyang2016 — APPROVED** [review]

**Review by nareddyt — APPROVED** [review]

File: tests/integration_test/backend_circuit_breaker_test/backend_circuit_breaker_test.go (hunk 3 above)
Comment (qiwzhang) [reply tests/integration_test/backend_circuit_breaker_test/backend_circuit_breaker_test.go:94]: Done

**Review by shuoyang2016 — APPROVED** [review]

**Review by TAOXUY — APPROVED** [review]

**Pull request merged**